content/docs/reference-react-dom-server.md

```diff
@@ -17,93 +17,89 @@ var ReactDOMServer = require('react-dom/server');
 
-```javascript
-ReactDOMServer.renderToStaticMarkup(element)
-```
+- [`renderToReadableStream()`](#rendertoreadablestream)
 
-Similar to [`renderToString`](#rendertostring), except this doesn't create extra DOM attributes that React uses internally, such as `data-reactroot`. This is useful if you want to use React as a simple static page generator, as stripping away the extra attributes can save some bytes.
+The following methods can be used in both the server and browser environments:
 
-If you plan to use React on the client to make the markup interactive, do not use this method. Instead, use [`renderToString`](#rendertostring) on the server and [`ReactDOM.hydrateRoot()`](/docs/react-dom-client.html#hydrateroot) on the client.
+- [`renderToString()`](#rendertostring)
+- [`renderToStaticMarkup()`](#rendertostaticmarkup)
 
-* * *
+## Reference {#reference}
 
 ### `renderToPipeableStream()` {#rendertopipeablestream}
 
 ```javascript
 ReactDOMServer.renderToPipeableStream(element, options)
 ```
 
-Render a React element to its initial HTML. Returns a [Control object](https://github.com/facebook/react/blob/3f8990898309c61c817fbf663f5221d9a00d0eaa/packages/react-dom/src/server/ReactDOMFizzServerNode.js#L49-L54) that allows you to pipe the output or abort the request. Fully supports Suspense and streaming of HTML with "delayed" content blocks "popping in" later through javascript execution. [Read more](https://github.com/reactwg/react-18/discussions/37)
+Render a React element to its initial HTML. Returns a stream with a `pipe(res)` method to pipe the output and `abort()` to abort the request. Fully supports Suspense and streaming of HTML with "delayed" content blocks "popping in" via inline `<script>` tags later. [Read more](https://github.com/reactwg/react-18/discussions/37)
 
 If you call [`ReactDOM.hydrateRoot()`](/docs/react-dom-client.html#hydrateroot) on a node that already has this server-rendered markup, React will preserve it and only attach event handlers, allowing you to have a very performant first-load experience.
 
-> Note:
->
-> This is a Node.js specific API and modern server environments should use renderToReadableStream instead.
->
-
-```
-const {pipe, abort} = renderToPipeableStream(
+```javascript
+let didError = false;
+const stream = renderToPipeableStream(
   <App />,
   {
-    onAllReady() {
-      res.statusCode = 200;
+    onShellReady() {
+      // The content above all Suspense boundaries is ready.
+      // If something errored before we started streaming, we set the error code appropriately.
+      res.statusCode = didError ? 500 : 200;
       res.setHeader('Content-type', 'text/html');
-      pipe(res);
+      stream.pipe(res);
     },
-    onShellError(x) {
+    onShellError(error) {
+      // Something errored before we could complete the shell so we emit an alternative shell.
```

Comment:
I find this comment counter-intuitive. If this is where errors "before the shell" go, then why does "on complete shell" check errors from "on error"?

Reply:
It doesn't have to but technically you could have had an error inside a suspense boundary before the shell completed and you can use that to tag the status code. However, that's not guaranteed to be sufficient.

Reply:
I'm actually not sure it's ever best-practice to set a 500 status code. You'd think that maybe it would be good for SEO bots but it's actually worse. Because they stop reindexing so aggressively. So instead of trying again when it might be fixed, it tries less.

If there is already something in the index maybe it prevents it from being overridden but I'm not sure and if it's the first time it's not good.

Also if the error is small, maybe it's still better to index what you successfully did.

Reply:
We should probably show what people to do today since we're just trying to explain the API

Reply:
Yea, people seem to come to the conclusion that streaming is impossible/bad by following that logic though.

Reply:
ok just tell me what to put here :)

Reply:
or we could just ship this for the other changes since this part was already there

Reply:
Yea we can follow up